codegen/src/main/java/software/amazon/awssdk/codegen/docs/OperationDocProvider.java

```diff
@@ -172,9 +172,15 @@ final void emitRequestParm(DocumentationBuilder docBuilder) {
         String parameterDocs = stripHtmlTags(opModel.getInput().getDocumentation());
 
         if (config.isConsumerBuilder()) {
+
+            String inputShapeName = opModel.getInputShape().getC2jName();
+            if (inputShapeName.endsWith("Input")) {
+                inputShapeName = inputShapeName.replaceAll("Input$", "Request");
+            }
+
             docBuilder.param(opModel.getInput().getVariableName(),
                              "A {@link Consumer} that will call methods on {@link %s.Builder} to create a request. %s",
-                             opModel.getInputShape().getC2jName(),
```

Comment:
Can we put the fully qualified name here instead?

Reply:
Sorry I meant the FQCN of the class; so for example `software.amazon.awssdk.s3.GetObject` or something like that. You should be able to use something like https://github.com/aws/aws-sdk-java-v2/blob/598237302a1089c8974ce91196b9c20c9e61282b/codegen/src/main/java/software/amazon/awssdk/codegen/poet/PoetExtension.java#L41 to do that (You'll need to get a reference to `IntermediateModel`)

Reply:
gotchu, updated to use fqcn